src/files/infra/repositories/FilesRepository.ts

```diff
@@ -0,0 +1,92 @@
+import { ApiRepository } from '../../../core/infra/repositories/ApiRepository';
+import { IFilesRepository } from '../../domain/repositories/IFilesRepository';
+import { FileOrderCriteria } from '../../domain/models/FileOrderCriteria';
+import { File } from '../../domain/models/File';
+import { transformFilesResponseToFiles } from './transformers/fileTransformers';
+import { FileDataTable } from '../../domain/models/FileDataTable';
+import { transformDataTablesResponseToDataTables } from './transformers/fileDataTableTransformers';
+import { FileUserPermissions } from '../../domain/models/FileUserPermissions';
+import { transformFileUserPermissionsResponseToFileUserPermissions } from './transformers/fileUserPermissionsTransformers';
+
+export interface GetFilesQueryParams {
+  limit?: number;
+  offset?: number;
+  orderCriteria?: string;
+}
+
+export class FilesRepository extends ApiRepository implements IFilesRepository {
+  public async getDatasetFiles(
+    datasetId: number | string,
+    datasetVersionId?: string,
+    limit?: number,
+    offset?: number,
+    orderCriteria?: FileOrderCriteria,
+  ): Promise<File[]> {
+    if (datasetVersionId === undefined) {
+      datasetVersionId = this.DATASET_VERSION_LATEST;
+    }
+    let endpoint;
+    if (typeof datasetId === 'number') {
+      endpoint = `/datasets/${datasetId}/versions/${datasetVersionId}/files`;
+    } else {
+      endpoint = `/datasets/:persistentId/versions/${datasetVersionId}/files?persistentId=${datasetId}`;
+    }
+    const queryParams: GetFilesQueryParams = {};
+    if (limit !== undefined) {
+      queryParams.limit = limit;
+    }
+    if (offset !== undefined) {
+      queryParams.offset = offset;
+    }
+    if (orderCriteria !== undefined) {
+      queryParams.orderCriteria = orderCriteria.toString();
+    }
```

Comment:
Yes, but it's significantly less readable and I preferred to use a cleaner approach since there aren't too many parameters, as in the example you shared with me from a "transformer" function.